src/layouts/Dashboard/components/ReviewRequestModal/RequestOverview/RequestOverview.tsx

```diff
@@ -290,37 +366,43 @@ export const RequestOverview = ({
           ref={inputRef}
           onSearchIconClick={handleExpansion}
           isExpanded={isExpanded}
-          onSearch={() => {
-            console.log("hi")
-          }}
+          onSearch={table.getColumn("itemName").setFilterValue}
           onBlur={handleCollapse}
         />
       </Flex>
       <Box w="100%" borderWidth="1px" borderRadius="8px" borderColor="gray.100">
         <TableVirtuoso
-          style={{ height: 400, borderRadius: "8px" }}
-          data={items}
+          style={{
+            height: 400,
+            borderRadius: "8px",
+          }}
+          // NOTE: Pass in only the list of filtered rows.
+          // This is to ensure that no indexing error occurs
+          // Eg: we filter so that only 1 item remains but render index is >1
+          data={table.getFilteredRowModel().rows}
           totalCount={items.length}
           components={{
             Table,
             TableBody: Tbody,
             TableRow: Tr,
             TableHead: (props) => <Thead {...props} bg="neutral.100" />,
           }}
-          fixedHeaderContent={() => (
-            <Tr>
-              {table
-                .getFlatHeaders()
-                .map((header) =>
-                  flexRender(
+          fixedHeaderContent={() =>
```

Comment:
I think the table header row is missing a bottom border.

Reply:
yea... i tried adding it in by setting a `borderBottom` on both `Th` and `Tr` but neither of them worked. the table content actually overlays the header so it hides teh border, annoyingly enough. checked w/ @NatMaeTan on this and she's fine w/ this but if you want, we could file an issue and i could work on this after!

Reply:
If design is ok I'm fine too!

Reply:
Idk what this is but if I told chin it's fine it should be ok